dbeam-core/src/main/java/com/spotify/dbeam/options/KmsSecrets.java

```diff
@@ -0,0 +1,185 @@
+/*-
+ * -\-\-
+ * DBeam Core
+ * --
+ * Copyright (C) 2016 - 2018 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.dbeam.options;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.util.Utils;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.services.cloudkms.v1.CloudKMS;
+import com.google.api.services.cloudkms.v1.CloudKMSScopes;
+import com.google.api.services.cloudkms.v1.model.DecryptRequest;
+import com.google.api.services.cloudkms.v1.model.DecryptResponse;
+import com.google.auto.value.AutoValue;
+import com.google.cloud.ServiceOptions;
+import com.google.common.base.CharMatcher;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Optional;
+import java.util.Properties;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KmsSecrets {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(PasswordReader.class);
+  private static final String DEFAULT_KEYRING;
+  private static final String DEFAULT_KEY;
+  private static final String DEFAULT_LOCATION;
+
+  static {
+    Properties p = System.getProperties();
+    DEFAULT_KEYRING = p.getProperty("KMS_DEFAULT_KEYRING", "dbeam");
+    DEFAULT_KEY = p.getProperty("KMS_DEFAULT_KEY", "default");
+    DEFAULT_LOCATION = p.getProperty("KMS_DEFAULT_LOCATION", "global");
+  }
+
+  /**
+   * Create a new {@link Decrypter.Builder} for customizing the {@link Decrypter} configuration.
+   */
+  public static Decrypter.Builder decrypter() {
+    Decrypter.Builder decrypter = new AutoValue_KmsSecrets_Decrypter.Builder()
+        .location(DEFAULT_LOCATION)
+        .key(DEFAULT_KEY)
+        .keyring(DEFAULT_KEYRING)
+        .project(ServiceOptions.getDefaultProjectId());
```

Comment:
I remember this has some unexpected behaviour. We might want to make it configuration as well.